File: frontend/src/api/systemPrompt.ts
import apiClient from './client'

export interface SystemPromptItem {
  id: number
  content: string
  updated_at: string | null
}

export interface SystemPromptUpdateResponse {
  success: boolean
  message: string
}

export interface SystemPromptResetResponse {
  success: boolean
  message: string
  content: string
}

export const systemPromptApi = {
  /** 获取当前启用的系统提示词 */
  get() {
    return apiClient.get<SystemPromptItem>('/system-prompt/')
  },

  /** 更新系统提示词 */
  update(content: string) {
    return apiClient.put<SystemPromptUpdateResponse>('/system-prompt/', { content })
  },

  /** 重置为默认系统提示词 */
  reset() {
    return apiClient.post<SystemPromptResetResponse>('/system-prompt/reset')
  },
}

File: backend/app/schemas/system_prompt.py
# System Prompt Schemas
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, field_validator


class SystemPromptOut(BaseModel):
    """Response schema for GET /api/system-prompt"""
    id: int
    content: str
    updated_at: Optional[datetime] = None


class SystemPromptUpdate(BaseModel):
    """Request schema for PUT /api/system-prompt"""
    content: str

    @field_validator("content")
    @classmethod
    def content_not_empty(cls, v: str) -> str:
        if not v or not v.strip():
            raise ValueError("提示词内容不能为空")
        return v.strip()


class SystemPromptResetResponse(BaseModel):
    """Response schema for POST /api/system-prompt/reset"""
    success: bool
    message: str
    content: str


class SystemPromptUpdateResponse(BaseModel):
    """Response schema for PUT /api/system-prompt"""
    success: bool
    message: str